contrib/go/src/python/pants/contrib/go/targets/go_target.py

```diff
@@ -65,3 +65,10 @@ def import_path(self):
     :returns: An import path that can be used to import this package in a `.go` file.
     :rtype: string
     """
+
+  def __init__(self, build_flags=None, **kwargs):
+    """
+    :param string build_flags: Build flags to pass to Go compiler.
+    """
+    super(GoTarget, self).__init__(**kwargs)
+    self.build_flags = build_flags
```

Comment:
IIUC these should only be accepted for `GoBinary` targets and they should invalidate compiles (installs). There are 2 paths to invalidation:
1. Include field into the default fingerprint of the target. An example is in [`GoLocalSource`], although you just need `add_field` singular + a `PrimitiveField` (https://github.com/pantsbuild/pants/blob/master/contrib/go/src/python/pants/contrib/go/targets/go_local_source.py#L67)
2. Use a custom `FingerprintStrategy` in the task(s) that cares about this field. A FingerprintEStrategy example is [here](https://github.com/pantsbuild/pants/blob/master/contrib/scrooge/src/python/pants/contrib/scrooge/tasks/java_thrift_library_fingerprint_strategy.py#L33) and and you'd pass it in as the custom `fingerprint_strategy` in the `GoCompile` [invalidated block](https://github.com/pantsbuild/pants/blob/master/contrib/go/src/python/pants/contrib/go/tasks/go_compile.py#L39).

I think you want 2 here since build flags are only used by `GoCompile` now and conceivably forever. In addition, the fingerprinting probably ought to be over the final `build_flags` you compose in `GoCompile` since order matter in the case of dups. This probably also means taking the `fingerprint=True` off the option (and maybe adding a comment that fp is handled by the custom FPStrategy) and bumping the task `implementation_version`.